src/Text/index.js

```diff
@@ -3,22 +3,65 @@
 import * as React from 'react';
 import { Text as RNText } from 'react-native';
 
-import { StyleSheet } from '..';
+import { StyleSheet } from '../index';
+import { alingGen, colorGen, fontSizeGen, fontWeightGen } from './utils';
 
-type Props = {
-  children: React.Node,
-};
+type TextProps = {|
+  +size?: 'normal' | 'large' | 'small',
+  +children: React.Node,
+  +uppercase?: boolean,
+  +italic?: boolean,
+  +align?: 'left' | 'right' | 'center' | 'justify',
+  +fontWeight?: 'normal' | 'bold',
+  +type?:
+    | 'primary'
+    | 'secondary'
+    | 'attention'
+    | 'info'
+    | 'success'
+    | 'warning'
+    | 'critical'
+    | 'white',
+|};
 
-const Text = ({ children }: Props) => (
-  <RNText style={styles.title}>{children}</RNText>
+const Text = ({
+  children,
+  italic,
+  uppercase,
+  align = 'left',
+  type = 'primary',
+  fontWeight = 'normal',
+  size = 'medium',
+}: TextProps) => (
+  <RNText
+    style={[
+      styles.text,
+      italic && styles.italic,
+      uppercase && styles.uppercase,
+      styles[fontWeight],
+      styles[size],
+      styles[type],
+      styles[align],
+    ]}
+  >
+    {children}
+  </RNText>
 );
 
 const styles = StyleSheet.create({
-  title: {
-    fontSize: 15,
-    fontWeight: 'bold',
-    color: 'green',
+  text: {
+    margin: 0,
   },
+  italic: {
+    fontStyle: 'italic',
+  },
+  uppercase: {
+    textTransform: 'uppercase',
+  },
+  ...fontWeightGen(),
```

Comment:
Maybe with the approach I used, we won't need to generate so many unused styles.

Reply:
Sure definitely I like your approach so I use your pattern.